src/adapter/optimizations/item_lookup.rs

```diff
@@ -264,15 +275,24 @@ pub(crate) fn resolve_crate_items<'a, V: AsVertex<Vertex<'a>> + 'a>(
 fn resolve_items_by_importable_path<'a>(
     crate_vertex: &'a IndexedCrate,
     origin: Origin,
+    destination_type: Option<Arc<str>>,
```

Comment:
This is a good addition, but now the function's contract is dissonant with its name: we're doing a lot more than just resolving items by importable path here. We're probably at the limit of how big the function's name can get, though.

Please add doc comments here and in the other function explaining the new post-filtering behavior.

Reply:
I've added comments in now, but I think they're a bit short - please let me know how you think they could be improved.